upgrader.go

```diff
@@ -172,12 +177,11 @@ func (u *Upgrader) handleUpgradeRequest(conn *net.UnixConn) {
 	u.l.Info("handling an upgrade request from peer")
 	u.Fds.lockMutations(ErrUpgradeInProgress)
 	// time to pass our FDs along
-	nextOwner, errC := passFdsToSibling(u.l, conn, u.Fds.copy())
+	conn.SetDeadline(u.clock.Now().Add(u.upgradeTimeout))
```

Comment:
The deadline change was made when I was still trying to figure out why `conn.Close()` didn't work (aka while I was still figuring out I'd leaked a file descriptor).

I can get rid of this in favor of the timer again, which is good because the timer can be mocked, and setdeadline can't be.